app/src/main/java/com/framgia/fsalon/wiget/StickyHeaderLayoutManager.java

```diff
@@ -0,0 +1,855 @@
+     * Updates firstViewAdapterPosition to the adapter position  of the highest item
+     * in the list - e.g., the
+     * adapter position of the item with lowest y value in the list
+     *
+     * @return the y value of the topmost view in the layout, or paddingTop if empty
+     */
+    int updateFirstAdapterPosition() {
+        // we're empty
+        if (getChildCount() == 0) {
+            firstViewAdapterPosition = 0;
+            firstViewTop = getPaddingTop();
+            return firstViewTop;
+        }
+        View topmostView = getTopmostChildView();
+        if (topmostView != null) {
+            firstViewAdapterPosition = getViewAdapterPosition(topmostView);
+            firstViewTop = Math.min(topmostView.getTop(), getPaddingTop());
+            return firstViewTop;
+        }
+        // as far as I can tell, if notifyDataSetChanged is called, onLayoutChildren
+        // will be called, but all views will be marked as having NO_POSITION for
+        // adapterPosition, which means the above approach of finding the topmostChildView
+        // doesn't work. So, basically, leave firstViewAdapterPosition and firstViewTop
+        // alone.
+        return firstViewTop;
+    }
+
+    void updateHeaderPositions(RecyclerView.Recycler recycler) {
+        // first, for each section represented by the current list of items,
+        // ensure that the header for that section is extant
+        Set<Integer> visitedSections = new HashSet<>();
+        for (int i = 0, n = getChildCount(); i < n; i++) {
+            View view = getChildAt(i);
+            int sectionIndex = getViewSectionIndex(view);
+            if (visitedSections.add(sectionIndex)) {
+                createSectionHeaderIfNeeded(recycler, sectionIndex);
+            }
+        }
+        // header is always positioned at top
+        int left = getPaddingLeft();
+        int right = getWidth() - getPaddingRight();
+        for (View headerView : headerViews) {
+            int sectionIndex = getViewSectionIndex(headerView);
+            // find first and last non-header views in this section
+            View ghostHeader = null;
+            View firstViewInNextSection = null;
+            for (int i = 0, n = getChildCount(); i < n; i++) {
+                View view = getChildAt(i);
+                // the view has been recycled
+                if (isViewRecycled(view)) {
+                    continue;
+                }
+                int type = getViewBaseType(view);
+                if (type == SectioningAdapter.TYPE_HEADER) {
+                    continue;
+                }
+                int viewSectionIndex = getViewSectionIndex(view);
+                if (viewSectionIndex == sectionIndex) {
+                    if (type == SectioningAdapter.TYPE_GHOST_HEADER) {
+                        ghostHeader = view;
+                    }
+                } else if (viewSectionIndex == sectionIndex + 1) {
+                    if (firstViewInNextSection == null) {
```

Comment:
WARNING: [CollapsibleIfStatements] These nested if statements could be combined
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/basic.html#CollapsibleIfStatements